cartographer/io/internal/mapping_state_serialization.cc

```diff
@@ -83,15 +83,13 @@ void SerializeSubmaps(
   SerializedData proto;
   // Next serialize all submaps.
   for (const auto& submap_id_data : submap_data) {
-    auto* const submap_proto = proto.mutable_submap();
-    submap_proto->mutable_submap_id()->set_trajectory_id(
-        submap_id_data.id.trajectory_id);
-    submap_proto->mutable_submap_id()->set_submap_index(
-        submap_id_data.id.submap_index);
-    submap_id_data.data.submap->ToProto(
-        submap_proto,
+    *proto.mutable_submap() = submap_id_data.data.submap->ToProto(
```

Comment:
Consistency with `SerializeTrajectoryNodes()`: Extract a local variable for `proto.mutable_submap()`?